superset/views/dashboard/api.py

```diff
@@ -88,12 +88,14 @@ class BaseDashboardSchema(BaseOwnedSchema):
     @pre_load
     def pre_load(self, data):  # pylint: disable=no-self-use
         super().pre_load(data)
-        data["slug"] = data.get("slug")
-        data["owners"] = data.get("owners", [])
-        if data["slug"]:
-            data["slug"] = data["slug"].strip()
-            data["slug"] = data["slug"].replace(" ", "-")
-            data["slug"] = re.sub(r"[^\w\-]+", "", data["slug"])
+        if "slug" in data:
+            data["slug"] = data.get("slug")
+            if data["slug"]:
```

Comment:
If `"slug" in data`, aren't the next rows unnecessary? Seems like we could just replace these three rows with `if get("slug"):`

Reply:
not exactly, I'm need to make sure that:
- Exists a slug key?
- If the slug key exists does it have data?
- If the slug key exists and it has data, apply string tranformations

Reply:
Forget it! your right